From b28abac9604f73456070703823c57bf3d8aec658 Mon Sep 17 00:00:00 2001
From: Ludovic Desroches <ludovic.desroches@atmel.com>
Date: Thu, 10 Jan 2013 11:26:42 +0100
Subject: [PATCH 276/319] SAMA5: big rework on dts files

Try to minimize code duplication. It's probably not a final version!

Signed-off-by: Ludovic Desroches <ludovic.desroches@atmel.com>
---
 arch/arm/boot/dts/sama5d3.dtsi       |  199 +++++++++++++++++-----------------
 arch/arm/boot/dts/sama5d31ek.dts     |  159 +--------------------------
 arch/arm/boot/dts/sama5d31ek_pda.dts |  174 +----------------------------
 arch/arm/boot/dts/sama5d33ek.dts     |  164 +---------------------------
 arch/arm/boot/dts/sama5d33ek_pda.dts |  179 +-----------------------------
 arch/arm/boot/dts/sama5d34ek.dts     |  159 +--------------------------
 arch/arm/boot/dts/sama5d34ek_pda.dts |  174 +----------------------------
 arch/arm/boot/dts/sama5d35ek.dts     |  160 +--------------------------
 arch/arm/boot/dts/sama5d3cm.dtsi     |   12 +-
 arch/arm/boot/dts/sama5d3dm.dtsi     |   62 +++++++++++
 arch/arm/boot/dts/sama5d3mb.dtsi     |  168 ++++++++++++++++++++++++++++
 arch/arm/boot/dts/sama5d3pda.dtsi    |   76 +++++++++++++
 12 files changed, 449 insertions(+), 1237 deletions(-)
 create mode 100644 arch/arm/boot/dts/sama5d3dm.dtsi
 create mode 100644 arch/arm/boot/dts/sama5d3mb.dtsi
 create mode 100644 arch/arm/boot/dts/sama5d3pda.dtsi

diff --git a/arch/arm/boot/dts/sama5d3.dtsi b/arch/arm/boot/dts/sama5d3.dtsi
index 3b53365..f77c9a4 100644
--- a/arch/arm/boot/dts/sama5d3.dtsi
+++ b/arch/arm/boot/dts/sama5d3.dtsi
@@ -61,7 +61,7 @@
 				reg = <0xf0000000 0x600>;
 				interrupts = <21 4 0>;
 				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc_dat4_7>;
+				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_dat4_7>;
 				status = "disabled";
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -530,27 +530,27 @@
 					};
 					pinctrl_adc0_ad10: adc0_ad10 {
 						atmel,pins =
-							<3 30 0x1 0x0>;	/* PD30 periph A AD10 */
+							<3 30 0x1 0x0>;	/* PD30 periph A AD10, conflicts with PCK0 */
 					};
 					pinctrl_adc0_ad11: adc0_ad11 {
 						atmel,pins =
-							<3 31 0x1 0x0>;	/* PD31 periph A AD11 */
+							<3 31 0x1 0x0>;	/* PD31 periph A AD11, conflicts with PCK1 */
 					};
 				};
 
 				can0 {
 					pinctrl_can0_rx_tx: can0_rx_tx {
 						atmel,pins =
-							<3 14 0x3 0x0	/* PD14 periph C RX */
-							 3 15 0x3 0x0>;	/* PD15 periph C TX */
+							<3 14 0x3 0x0	/* PD14 periph C RX, conflicts with SCK0, SPI0_NPCS1 */
+							 3 15 0x3 0x0>;	/* PD15 periph C TX, conflicts with CTS0, SPI0_NPCS2 */
 					};
 				};
 
 				can1 {
 					pinctrl_can1_rx_tx: can1_rx_tx {
 						atmel,pins =
-							<1 14 0x2 0x0	/* PB14 periph B RX */
-							 1 15 0x2 0x0>;	/* PB15 periph B TX */
+							<1 14 0x2 0x0	/* PB14 periph B RX, conflicts with GCRS */
+							 1 15 0x2 0x0>;	/* PB15 periph B TX, conflicts with GCOL */
 					};
 				};
 
@@ -565,45 +565,40 @@
 				i2c0 {
 					pinctrl_i2c0: i2c0-0 {
 						atmel,pins =
-							<0 30 0x1 0x0	/* PA30 periph A TWD0 pin */
-							 0 31 0x1 0x0>;	/* PA31 periph A TWCK0 pin */
+							<0 30 0x1 0x0	/* PA30 periph A TWD0 pin, conflicts with URXD1, ISI_VSYNC */
+							 0 31 0x1 0x0>;	/* PA31 periph A TWCK0 pin, conflicts with UTXD1, ISI_HSYNC */
 					};
 				};
 
 				i2c1 {
 					pinctrl_i2c1: i2c1-0 {
 						atmel,pins =
-							<2 26 0x2 0x0	/* PC26 periph B TWD1 pin */
-							 2 27 0x2 0x0>;	/* PC27 periph B TWCK1 pin */
+							<2 26 0x2 0x0	/* PC26 periph B TWD1 pin, conflicts with SPI1_NPCS1, ISI_D11 */
+							 2 27 0x2 0x0>;	/* PC27 periph B TWCK1 pin, conflicts with SPI1_NPCS2, ISI_D10 */
 					};
 				};
 
 				isi {
 					pinctrl_isi: isi-0 {
 						atmel,pins =
-							<0 16 0x3 0x0	/* PA16 periph C ISI_D0 */
-							 0 17 0x3 0x0	/* PA17 periph C ISI_D1 */
-							 0 18 0x3 0x0	/* PA18 periph C ISI_D2 */
-							 0 19 0x3 0x0	/* PA19 periph C ISI_D3 */
-							 0 20 0x3 0x0	/* PA20 periph C ISI_D4 */
-							 0 21 0x3 0x0	/* PA21 periph C ISI_D5 */
-							 0 22 0x3 0x0	/* PA22 periph C ISI_D6 */
-							 0 23 0x3 0x0	/* PA23 periph C ISI_D7 */
-							 2 30 0x3 0x0	/* PC30 periph C ISI_PCK */
-							 0 31 0x3 0x0	/* PA31 periph C ISI_HSYNC */
-							 0 30 0x3 0x0	/* PA30 periph C ISI_VSYMC */
-							 2 29 0x3 0x0	/* PC29 periph C ISI_PD8 */
-							 2 28 0x3 0x0>;	/* PC28 periph C ISI_PD9 */
+							<0 16 0x3 0x0	/* PA16 periph C ISI_D0, conflicts with LCDDAT16 */
+							 0 17 0x3 0x0	/* PA17 periph C ISI_D1, conflicts with LCDDAT17 */
+							 0 18 0x3 0x0	/* PA18 periph C ISI_D2, conflicts with LCDDAT18, TWD2 */
+							 0 19 0x3 0x0	/* PA19 periph C ISI_D3, conflicts with LCDDAT19, TWCK2 */
+							 0 20 0x3 0x0	/* PA20 periph C ISI_D4, conflicts with LCDDAT20, PWMH0 */
+							 0 21 0x3 0x0	/* PA21 periph C ISI_D5, conflicts with LCDDAT21, PWML0 */
+							 0 22 0x3 0x0	/* PA22 periph C ISI_D6, conflicts with LCDDAT22, PWMH1 */
+							 0 23 0x3 0x0	/* PA23 periph C ISI_D7, conflicts with LCDDAT23, PWML1 */
+							 2 30 0x3 0x0	/* PC30 periph C ISI_PCK, conflicts with UTXD0 */
+							 0 31 0x3 0x0	/* PA31 periph C ISI_HSYNC, conflicts with TWCK0, UTXD1 */
+							 0 30 0x3 0x0	/* PA30 periph C ISI_VSYNC, conflicts with TWD0, URXD1 */
+							 2 29 0x3 0x0	/* PC29 periph C ISI_PD8, conflicts with URXD0, PWMFI2 */
+							 2 28 0x3 0x0>;	/* PC28 periph C ISI_PD9, conflicts with SPI1_NPCS3, PWMFI0 */
 					};
 					pinctrl_isi_pck_as_mck: isi_pck_as_mck-0 {
 						atmel,pins =
 							<3 31 0x2 0x0>;	/* PD31 periph B ISI_MCK */
 					};
-					pinctrl_isi_camera: isi_camera-0 {
-						atmel,pins =
-							<4 24 0x0 0x0	/* PE24 gpio ZB_SLPTR */
-							 4 29 0x0 0x0>;	/* PE29 gpio ZB_RSTN */
-					};
 				};
 
 				lcd {
@@ -645,45 +640,45 @@
 				macb0 {
 					pinctrl_macb0_data_rgmii: macb0_data_rgmii {
 						atmel,pins =
-							<1 0 0x1 0x0	/* PB0 periph A GTX0 */
-							 1 1 0x1 0x0	/* PB1 periph A GTX1 */
-							 1 2 0x1 0x0	/* PB2 periph A GTX2 */
-							 1 3 0x1 0x0	/* PB3 periph A GTX3 */
-							 1 4 0x1 0x0	/* PB4 periph A GRX0 */
-							 1 5 0x1 0x0	/* PB5 periph A GRX1 */
-							 1 6 0x1 0x0	/* PB6 periph A GRX2 */
-							 1 7 0x1 0x0>;	/* PB7 periph A GRX3 */
+							<1 0 0x1 0x0	/* PB0 periph A GTX0, conflicts with PWMH0 */
+							 1 1 0x1 0x0	/* PB1 periph A GTX1, conflicts with PWML0 */
+							 1 2 0x1 0x0	/* PB2 periph A GTX2, conflicts with TK1 */
+							 1 3 0x1 0x0	/* PB3 periph A GTX3, conflicts with TF1 */
+							 1 4 0x1 0x0	/* PB4 periph A GRX0, conflicts with PWMH1 */
+							 1 5 0x1 0x0	/* PB5 periph A GRX1, conflicts with PWML1 */
+							 1 6 0x1 0x0	/* PB6 periph A GRX2, conflicts with TD1 */
+							 1 7 0x1 0x0>;	/* PB7 periph A GRX3, conflicts with RK1 */
 					};
 					pinctrl_macb0_data_gmii: macb0_data_gmii {
 						atmel,pins =
-							<1 19 0x2 0x0	/* PB19 periph B GTX4 */
-							 1 20 0x2 0x0	/* PB20 periph B GTX5 */
-							 1 21 0x2 0x0	/* PB21 periph B GTX6 */
-							 1 22 0x2 0x0	/* PB22 periph B GTX7 */
-							 1 23 0x2 0x0	/* PB23 periph B GRX4 */
-							 1 24 0x2 0x0	/* PB24 periph B GRX5 */
-							 1 25 0x2 0x0	/* PB25 periph B GRX6 */
-							 1 26 0x2 0x0>;	/* PB26 periph B GRX7 */
+							<1 19 0x2 0x0	/* PB19 periph B GTX4, conflicts with MCI1_CDA */
+							 1 20 0x2 0x0	/* PB20 periph B GTX5, conflicts with MCI1_DA0 */
+							 1 21 0x2 0x0	/* PB21 periph B GTX6, conflicts with MCI1_DA1 */
+							 1 22 0x2 0x0	/* PB22 periph B GTX7, conflicts with MCI1_DA2 */
+							 1 23 0x2 0x0	/* PB23 periph B GRX4, conflicts with MCI1_DA3 */
+							 1 24 0x2 0x0	/* PB24 periph B GRX5, conflicts with MCI1_CK */
+							 1 25 0x2 0x0	/* PB25 periph B GRX6, conflicts with SCK1 */
+							 1 26 0x2 0x0>;	/* PB26 periph B GRX7, conflicts with CTS1 */
 					};
 					pinctrl_macb0_signal_rgmii: macb0_signal_rgmii {
 						atmel,pins =
-							<1 8 0x1 0x0	/* PB8 periph A GTXCK */
-							 1 9 0x1 0x0	/* PB9 periph A GTXEN */
-							 1 11 0x1 0x0	/* PB11 periph A GRXCK */
-							 1 13 0x1 0x0	/* PB13 periph A GRXER */
+							<1 8 0x1 0x0	/* PB8 periph A GTXCK, conflicts with PWMH2 */
+							 1 9 0x1 0x0	/* PB9 periph A GTXEN, conflicts with PWML2 */
+							 1 11 0x1 0x0	/* PB11 periph A GRXCK, conflicts with RD1 */
+							 1 13 0x1 0x0	/* PB13 periph A GRXER, conflicts with PWML3 */
 							 1 16 0x1 0x0	/* PB16 periph A GMDC */
 							 1 17 0x1 0x0	/* PB17 periph A GMDIO */
 							 1 18 0x1 0x0>;	/* PB18 periph A G125CK */
 					};
 					pinctrl_macb0_signal_gmii: macb0_signal_gmii {
 						atmel,pins =
-							<1 9 0x1 0x0	/* PB9 periph A GTXEN */
-							 1 10 0x1 0x0	/* PB10 periph A GTXER */
-							 1 11 0x1 0x0	/* PB11 periph A GRXCK */
-							 1 12 0x1 0x0	/* PB12 periph A GRXDV */
-							 1 13 0x1 0x0	/* PB13 periph A GRXER */
-							 1 14 0x1 0x0	/* PB14 periph A GCRS */
-							 1 15 0x1 0x0	/* PB15 periph A GCOL */
+							<1 9 0x1 0x0	/* PB9 periph A GTXEN, conflicts with PWML2 */
+							 1 10 0x1 0x0	/* PB10 periph A GTXER, conflicts with RF1 */
+							 1 11 0x1 0x0	/* PB11 periph A GRXCK, conflicts with RD1 */
+							 1 12 0x1 0x0	/* PB12 periph A GRXDV, conflicts with PWMH3 */
+							 1 13 0x1 0x0	/* PB13 periph A GRXER, conflicts with PWML3 */
+							 1 14 0x1 0x0	/* PB14 periph A GCRS, conflicts with CANRX1 */
+							 1 15 0x1 0x0	/* PB15 periph A GCOL, conflicts with CANTX1 */
 							 1 16 0x1 0x0	/* PB16 periph A GMDC */
 							 1 17 0x1 0x0	/* PB17 periph A GMDIO */
 							 1 27 0x2 0x0>;	/* PB27 periph B G125CKO */
@@ -694,16 +689,16 @@
 				macb1 {
 					pinctrl_macb1_rmii: macb1_rmii-0 {
 						atmel,pins =
-							<2 0 0x1 0x0	/* PC0 periph A TX0   */
-							 2 1 0x1 0x0	/* PC1 periph A TX1   */
-							 2 2 0x1 0x0	/* PC2 periph A RX0   */
-							 2 3 0x1 0x0	/* PC3 periph A RX1   */
-							 2 4 0x1 0x0	/* PC4 periph A TXEN  */
-							 2 5 0x1 0x0	/* PC5 periph A CRSDV */
-							 2 6 0x1 0x0	/* PC6 periph A RXER  */
-							 2 7 0x1 0x0	/* PC7 periph A REFCK */
-							 2 8 0x1 0x0	/* PC8 periph A MDC   */
-							 2 9 0x1 0x0>;	/* PC9 periph A MDIO  */
+							<2 0 0x1 0x0	/* PC0 periph A ETX0, conflicts with TIOA3 */
+							 2 1 0x1 0x0	/* PC1 periph A ETX1, conflicts with TIOB3 */
+							 2 2 0x1 0x0	/* PC2 periph A ERX0, conflicts with TCLK3 */
+							 2 3 0x1 0x0	/* PC3 periph A ERX1, conflicts with TIOA4 */
+							 2 4 0x1 0x0	/* PC4 periph A ETXEN, conflicts with TIOB4 */
+							 2 5 0x1 0x0	/* PC5 periph A ECRSDV,conflicts with TCLK4 */
+							 2 6 0x1 0x0	/* PC6 periph A ERXER, conflicts with TIOA5 */
+							 2 7 0x1 0x0	/* PC7 periph A EREFCK, conflicts with TIOB5 */
+							 2 8 0x1 0x0	/* PC8 periph A EMDC, conflicts with TCLK5 */
+							 2 9 0x1 0x0>;	/* PC9 periph A EMDIO  */
 					};
 				};
 
@@ -722,40 +717,40 @@
 					};
 					pinctrl_mmc0_dat4_7: mmc0_dat4_7 {
 						atmel,pins =
-							<3 5 0x1 0x1	/* PD5 periph A MCI0_DA4 with pullup */
-							 3 6 0x1 0x1	/* PD6 periph A MCI0_DA5 with pullup */
-							 3 7 0x1 0x1	/* PD7 periph A MCI0_DA6 with pullup */
-							 3 8 0x1 0x1>;	/* PD8 periph A MCI0_DA7 with pullup */
+							<3 5 0x1 0x1	/* PD5 periph A MCI0_DA4 with pullup, conflicts with TIOA0, PWMH2 */
+							 3 6 0x1 0x1	/* PD6 periph A MCI0_DA5 with pullup, conflicts with TIOB0, PWML2 */
+							 3 7 0x1 0x1	/* PD7 periph A MCI0_DA6 with pullup, conlicts with TCLK0, PWMH3 */
+							 3 8 0x1 0x1>;	/* PD8 periph A MCI0_DA7 with pullup, conflicts with PWML3 */
 					};
 				};
 
 				mmc1 {
 					pinctrl_mmc1_clk_cmd_dat0: mmc1_clk_cmd_dat0 {
 						atmel,pins =
-							<1 24 0x1 0x0	/* PB24 periph A MCI1_CK */
-							 1 19 0x1 0x1	/* PB19 periph A MCI1_CDA with pullup */
-							 1 20 0x1 0x1>;	/* PB20 periph A MCI1_DA0 with pullup */
+							<1 24 0x1 0x0	/* PB24 periph A MCI1_CK, conflicts with GRX5 */
+							 1 19 0x1 0x1	/* PB19 periph A MCI1_CDA with pullup, conflicts with GTX4 */
+							 1 20 0x1 0x1>;	/* PB20 periph A MCI1_DA0 with pullup, conflicts with GTX5 */
 					};
 					pinctrl_mmc1_dat1_3: mmc1_dat1_3 {
 						atmel,pins =
-							<1 21 0x1 0x1	/* PB21 periph A MCI1_DA1 with pullup */
-							 1 22 0x1 0x1	/* PB22 periph A MCI1_DA2 with pullup */
-							 1 23 0x1 0x1>;	/* PB23 periph A MCI1_DA3 with pullup */
+							<1 21 0x1 0x1	/* PB21 periph A MCI1_DA1 with pullup, conflicts with GTX6 */
+							 1 22 0x1 0x1	/* PB22 periph A MCI1_DA2 with pullup, conflicts with GTX7 */
+							 1 23 0x1 0x1>;	/* PB23 periph A MCI1_DA3 with pullup, conflicts with GRX4 */
 					};
 				};
 
 				mmc2 {
 					pinctrl_mmc2_clk_cmd_dat0: mmc2_clk_cmd_dat0 {
 						atmel,pins =
-							<2 15 0x1 0x0	/* PC15 periph A MCI2_CK */
+							<2 15 0x1 0x0	/* PC15 periph A MCI2_CK, conflicts with PCK2 */
 							 2 10 0x1 0x1	/* PC10 periph A MCI2_CDA with pullup */
 							 2 11 0x1 0x1>;	/* PC11 periph A MCI2_DA0 with pullup */
 					};
 					pinctrl_mmc2_dat1_3: mmc2_dat1_3 {
 						atmel,pins =
-							<2 12 0x1 0x0	/* PC12 periph A MCI2_DA1 with pullup */
-							 2 13 0x1 0x0	/* PC13 periph A MCI2_DA2 with pullup */
-							 2 14 0x1 0x0>;	/* PC14 periph A MCI2_DA3 with pullup */
+							<2 12 0x1 0x0	/* PC12 periph A MCI2_DA1 with pullup, conflicts with TIOA1 */
+							 2 13 0x1 0x0	/* PC13 periph A MCI2_DA2 with pullup, conflicts with TIOB1 */
+							 2 14 0x1 0x0>;	/* PC14 periph A MCI2_DA3 with pullup, conflicts with TCLK1 */
 					};
 				};
 
@@ -856,32 +851,32 @@
 				ssc1 {
 					pinctrl_ssc1_tx: ssc1_tx {
 						atmel,pins =
-							<1 2 0x2 0x0	/* PB2 periph B TK1 */
-							 1 3 0x2 0x0	/* PB3 periph B TF1 */
-							 1 6 0x2 0x0>;	/* PB6 periph B TD1 */
+							<1 2 0x2 0x0	/* PB2 periph B TK1, conflicts with GTX2 */
+							 1 3 0x2 0x0	/* PB3 periph B TF1, conflicts with GTX3 */
+							 1 6 0x2 0x0>;	/* PB6 periph B TD1, conflicts with TD1 */
 					};
 
 					pinctrl_ssc1_rx: ssc1_rx {
 						atmel,pins =
-							<1 7 0x2 0x0	/* PC7 periph B RK1 */
-							 1 10 0x2 0x0	/* PC10 periph B RF1 */
-							 1 11 0x2 0x0>;	/* PC11 periph B RD1 */
+							<1 7 0x2 0x0	/* PB7 periph B RK1, conflicts with EREFCK */
+							 1 10 0x2 0x0	/* PB10 periph B RF1, conflicts with GTXER */
+							 1 11 0x2 0x0>;	/* PB11 periph B RD1, conflicts with GRXCK */
 					};
 				};
 
 				uart0 {
 					pinctrl_uart0: uart0-0 {
 						atmel,pins =
-							<3 29 0x1 0x0	/* PC29 periph A */
-							 3 30 0x1 0x1>;	/* PC30 periph A with pullup */
+							<2 29 0x1 0x0	/* PC29 periph A, conflicts with PWMFI2, ISI_D8 */
+							 2 30 0x1 0x1>;	/* PC30 periph A with pullup, conflicts with ISI_PCK */
 					};
 				};
 
 				uart1 {
 					pinctrl_uart1: uart1-0 {
 						atmel,pins =
-							<0 30 0x2 0x0	/* PA30 periph B */
-							 0 31 0x2 0x1>;	/* PA31 periph B with pullup */
+							<0 30 0x2 0x0	/* PA30 periph B, conflicts with TWD0, ISI_VSYNC */
+							 0 31 0x2 0x1>;	/* PA31 periph B with pullup, conflicts with TWCK0, ISI_HSYNC */
 					};
 				};
 
@@ -894,8 +889,8 @@
 
 					pinctrl_usart0_rts_cts: usart0_rts_cts-0 {
 						atmel,pins =
-							<3 15 0x1 0x0	/* PD15 periph A */
-							 3 16 0x1 0x0>;	/* PD16 periph A */
+							<3 15 0x1 0x0	/* PD15 periph A, conflicts with SPI0_NPCS2, CANTX0 */
+							 3 16 0x1 0x0>;	/* PD16 periph A, conflicts with SPI0_NPCS3, PWMFI3 */
 					};
 				};
 
@@ -908,36 +903,36 @@
 
 					pinctrl_usart1_rts_cts: usart1_rts_cts-0 {
 						atmel,pins =
-							<1 26 0x1 0x0	/* PB26 periph A */
-							 1 27 0x1 0x0>;	/* PB27 periph A */
+							<1 26 0x1 0x0	/* PB26 periph A, conflicts with GRX7 */
+							 1 27 0x1 0x0>;	/* PB27 periph A, conflicts with G125CKO */
 					};
 				};
 
 				usart2 {
 					pinctrl_usart2: usart2-0 {
 						atmel,pins =
-							<4 25 0x2 0x0	/* PE25 periph B */
-							 4 26 0x2 0x1>;	/* PE26 periph B with pullup */
+							<4 25 0x2 0x0	/* PE25 periph B, conflicts with A25 */
+							 4 26 0x2 0x1>;	/* PE26 periph B with pullup, conflicts NCS0 */
 					};
 
 					pinctrl_usart2_rts_cts: usart2_rts_cts-0 {
 						atmel,pins =
-							<4 23 0x2 0x0	/* PE23 periph B */
-							 4 24 0x2 0x0>;	/* PE24 periph B */
+							<4 23 0x2 0x0	/* PE23 periph B, conflicts with A23 */
+							 4 24 0x2 0x0>;	/* PE24 periph B, conflicts with A24 */
 					};
 				};
 
 				usart3 {
 					pinctrl_usart3: usart3-0 {
 						atmel,pins =
-							<4 18 0x2 0x0	/* PE18 periph B */
-							 4 19 0x2 0x1>;	/* PE19 periph B with pullup */
+							<4 18 0x2 0x0	/* PE18 periph B, conflicts with A18 */
+							 4 19 0x2 0x1>;	/* PE19 periph B with pullup, conflicts with A19 */
 					};
 
 					pinctrl_usart3_rts_cts: usart3_rts_cts-0 {
 						atmel,pins =
-							<4 16 0x2 0x0	/* PE16 periph B */
-							 4 17 0x2 0x0>;	/* PE17 periph B */
+							<4 16 0x2 0x0	/* PE16 periph B, conflicts with A16 */
+							 4 17 0x2 0x0>;	/* PE17 periph B, conflicts with A17 */
 					};
 				};
 			};
diff --git a/arch/arm/boot/dts/sama5d31ek.dts b/arch/arm/boot/dts/sama5d31ek.dts
index 15e9b4e..37129f9 100644
--- a/arch/arm/boot/dts/sama5d31ek.dts
+++ b/arch/arm/boot/dts/sama5d31ek.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3dm.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D31-EK";
+	compatible = "atmel,sama5d31ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -45,158 +30,26 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
 				status = "okay";
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
-				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
-			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "okay";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
 			};
 
 			macb1: ethernet@f802c000 {
-				phy-mode = "rmii";
-				status = "okay";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>;	/* PE31 GPIO with pull up deglith */
-					};
-
-					pinctrl_usba_vbus: usba_vbus {
-						atmel,pins =
-							<3 29 0x0 0x4>; /* PD29 GPIO with deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
 				status = "okay";
 			};
 		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 0>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usba_vbus>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 0
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
-		};
 	};
 
 	leds {
 		d3 {
 			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
+			gpios = <&pioE 24 0>;
 		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d31ek_pda.dts b/arch/arm/boot/dts/sama5d31ek_pda.dts
index a988b2f..409e58e 100644
--- a/arch/arm/boot/dts/sama5d31ek_pda.dts
+++ b/arch/arm/boot/dts/sama5d31ek_pda.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3pda.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D31-EK_PDA";
+	compatible = "atmel,sama5d31ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -45,173 +30,26 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
 				status = "okay";
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <30 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-
-				atmel_mxt_ts@4a {
-					compatible = "atmel,atmel_mxt_ts";
-					reg = <0x4a>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_mxt_ts>;
-					x_line = <19>;
-					y_line = <11>;
-					x_size = <1024>;
-					y_size = <1024>;
-					blen = <0x0>;
-					threshold = <0x23>;
-					voltage = <3300000>;
-					orient = <0x00>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
-				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
-			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "disabled";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
 			};
 
 			macb1: ethernet@f802c000 {
-				phy-mode = "rmii";
-				status = "okay";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 30 0x0 0x5>;	/* PE30 GPIO with pull up deglith */
-					};
-
-					pinctrl_mxt_ts: mxt_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>; /* PE31 gpio with pullup deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
 				status = "okay";
 			};
 		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 1>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 1
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
-		};
 	};
 
 	leds {
 		d3 {
 			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
+			gpios = <&pioE 24 0>;
 		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d33ek.dts b/arch/arm/boot/dts/sama5d33ek.dts
index d45a59f..106454b 100644
--- a/arch/arm/boot/dts/sama5d33ek.dts
+++ b/arch/arm/boot/dts/sama5d33ek.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3dm.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D33-EK";
+	compatible = "atmel,sama5d33ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -45,158 +30,19 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
 				status = "okay";
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
 			};
 
 			macb0: ethernet@f0028000 {
-				phy-mode = "rgmii";
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
-				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
-			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "okay";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>;	/* PE31 GPIO with pull up deglith */
-					};
-
-					pinctrl_usba_vbus: usba_vbus {
-						atmel,pins =
-							<3 29 0x0 0x4>; /* PD29 GPIO with deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
 				status = "okay";
 			};
 		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 0>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usba_vbus>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 0
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
-		};
-	};
-
-	leds {
-		d3 {
-			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
-		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d33ek_pda.dts b/arch/arm/boot/dts/sama5d33ek_pda.dts
index 8ec54ad..a4d57eb5 100644
--- a/arch/arm/boot/dts/sama5d33ek_pda.dts
+++ b/arch/arm/boot/dts/sama5d33ek_pda.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3pda.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D33-EK_PDA";
+	compatible = "atmel,sama5d33ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -45,173 +30,19 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
 				status = "okay";
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <30 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-
-				atmel_mxt_ts@4a {
-					compatible = "atmel,atmel_mxt_ts";
-					reg = <0x4a>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_mxt_ts>;
-					x_line = <19>;
-					y_line = <11>;
-					x_size = <1024>;
-					y_size = <1024>;
-					blen = <0x0>;
-					threshold = <0x23>;
-					voltage = <3300000>;
-					orient = <0x00>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
 			};
 
 			macb0: ethernet@f0028000 {
-				phy-mode = "rgmii";
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
-				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
-			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "disabled";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 30 0x0 0x5>;	/* PE30 GPIO with pull up deglith */
-					};
-
-					pinctrl_mxt_ts: mxt_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>; /* PE31 gpio with pullup deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
 				status = "okay";
 			};
 		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 1>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 1
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
-		};
-	};
-
-	leds {
-		d3 {
-			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
-		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d34ek.dts b/arch/arm/boot/dts/sama5d34ek.dts
index 3735847..d14af2f 100644
--- a/arch/arm/boot/dts/sama5d34ek.dts
+++ b/arch/arm/boot/dts/sama5d34ek.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3dm.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D34-EK";
+	compatible = "atmel,sama5d34ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -49,11 +34,6 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
@@ -64,149 +44,22 @@
 					reg = <0x50>;
 					pagesize = <64>;
 				};
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
 			};
 
 			macb0: ethernet@f0028000 {
-				phy-mode = "rgmii";
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
 				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
 			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "okay";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>;	/* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>;	/* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>;	/* PE31 GPIO with pull up deglith */
-					};
-
-					pinctrl_usba_vbus: usba_vbus {
-						atmel,pins =
-							<3 29 0x0 0x4>; /* PD29 GPIO with deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
-				status = "okay";
-			};
-		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 0>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usba_vbus>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 0
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
 		};
 	};
 
 	leds {
 		d3 {
 			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
+			gpios = <&pioE 24 0>;
 		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d34ek_pda.dts b/arch/arm/boot/dts/sama5d34ek_pda.dts
index ec863dd..394e8c8 100644
--- a/arch/arm/boot/dts/sama5d34ek_pda.dts
+++ b/arch/arm/boot/dts/sama5d34ek_pda.dts
@@ -7,12 +7,12 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
+/include/ "sama5d3pda.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D34-EK_PDA";
+	compatible = "atmel,sama5d34ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,23 +20,8 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
 			};
 
 			ssc0: ssc@f0008000 {
@@ -49,11 +34,6 @@
 
 			i2c0: i2c@f0014000 {
 				status = "okay";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
 			};
 
 			i2c1: i2c@f0018000 {
@@ -64,164 +44,22 @@
 					reg = <0x50>;
 					pagesize = <64>;
 				};
-
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <30 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-
-				atmel_mxt_ts@4a {
-					compatible = "atmel,atmel_mxt_ts";
-					reg = <0x4a>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_mxt_ts>;
-					x_line = <19>;
-					y_line = <11>;
-					x_size = <1024>;
-					y_size = <1024>;
-					blen = <0x0>;
-					threshold = <0x23>;
-					voltage = <3300000>;
-					orient = <0x00>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
 			};
 
 			macb0: ethernet@f0028000 {
-				phy-mode = "rgmii";
-				status = "okay";
-			};
-
-			lcd_bus@f0030000 {
-				status = "okay";
-
-				lcd@f0030000 {
-					status = "okay";
-				};
-
-				lcdovl1@f0030140 {
-					status = "okay";
-				};
-
-				lcdovl2@f0030240 {
-					status = "okay";
-				};
-
-				lcdheo1@f0030340 {
-					status = "okay";
-				};
-			};
-
-			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "disabled";
-			};
-
-			mmc1: mmc@f8000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			tsadcc: tsadcc@f8018000 {
-				status = "disabled";
-			};
-
-			adc0: adc@f8018000 {
-				status = "disabled";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>;	/* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>;	/* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 30 0x0 0x5>;	/* PE30 GPIO with pull up deglith */
-					};
-
-					pinctrl_mxt_ts: mxt_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>; /* PE31 gpio with pullup deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
 				status = "okay";
 			};
-
-			watchdog@fffffe40 {
-				status = "okay";
-			};
-		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 1>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 1
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
 		};
 	};
 
 	leds {
 		d3 {
 			label = "d3";
-			gpios = <&pioE 24 0>;	/* PE24, conflicts with camera reset pin and RTS2 */
-			linux,default-trigger = "heartbeat";
+			gpios = <&pioE 24 0>;
 		};
 	};
 
 	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
+		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d35ek.dts b/arch/arm/boot/dts/sama5d35ek.dts
index 78e82d8..b196bdc 100644
--- a/arch/arm/boot/dts/sama5d35ek.dts
+++ b/arch/arm/boot/dts/sama5d35ek.dts
@@ -7,12 +7,11 @@
  * Licensed under GPLv2 or later.
  */
 /dts-v1/;
-/include/ "sama5d3.dtsi"
-/include/ "sama5d3cm.dtsi"
+/include/ "sama5d3mb.dtsi"
 
 / {
-	model = "Atmel SAMA5D3-EK";
-	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+	model = "Atmel SAMA5D35-EK";
+	compatible = "atmel,sama5d35ek", "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
 
 	chosen {
 		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
@@ -20,182 +19,29 @@
 
 	ahb {
 		apb {
-			mmc0: mmc@f0000000 {
-				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 17 0>;
-				};
-			};
-
 			spi0: spi@f0004000 {
 				status = "okay";
-				m25p80@0 {
-					compatible = "atmel,at25df321a";
-					spi-max-frequency = <50000000>;
-					reg = <0>;
-				};
-			};
-
-			ssc0: ssc@f0008000 {
-				status = "disabled";
 			};
 
 			can0: can@f000c000 {
 				status = "okay";
 			};
 
-			/*
-			 * i2c0 conflicts with ISI:
-			 * disable it to allow the use of ISI
-			 * can not enable audio when i2c0 disabled
-			 */
-			i2c0: i2c@f0014000 {
-				status = "disabled";
-
-				wm8904: wm8904@1a {
-					compatible = "wm8904";
-					reg = <0x1a>;
-				};
-			};
-
 			i2c1: i2c@f0018000 {
 				status = "okay";
-
-				/* No LCD support
-				 * not sure enable qt1070 make sense
-				 */
-				qt1070: keyboard@1b {
-					compatible = "qt1070";
-					reg = <0x1b>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 0x0>;
-					pinctrl-names = "default";
-					pinctrl-0 = <&pinctrl_qt1070_irq>;
-				};
-			};
-
-			usart1: serial@f0020000 {
-				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
-				status = "okay";
 			};
 
 			macb0: ethernet@f0028000 {
-				phy-mode = "rgmii";
 				status = "okay";
 			};
 
 			isi: isi@f0034000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_camera>;
-				status = "okay";
-			};
-
-			mmc1: mmc@f8000000 {
 				status = "okay";
-				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
-				slot@0 {
-					reg = <0>;
-					bus-width = <4>;
-					cd-gpios = <&pioD 18 0>;
-				};
-			};
-
-			adc0: adc@f8018000 {
-				status = "okay";
-				pinctrl-0 = <
-					&pinctrl_adc0_adtrg
-					&pinctrl_adc0_ad0
-					&pinctrl_adc0_ad1
-					&pinctrl_adc0_ad2
-					&pinctrl_adc0_ad3
-					&pinctrl_adc0_ad4
-					>;
 			};
 
 			macb1: ethernet@f802c000 {
-				phy-mode = "rmii";
-				status = "okay";
-			};
-
-			pinctrl@fffff200 {
-				board {
-					pinctrl_mmc0_cd: mmc0_cd {
-						atmel,pins =
-							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
-					};
-
-					pinctrl_mmc1_cd: mmc1_cd {
-						atmel,pins =
-							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
-					};
-
-					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
-						atmel,pins =
-							<3 30 0x2 0x0>;	/* PD30 periph B */
-					};
-
-					pinctrl_qt1070_irq: qt1070_irq {
-						atmel,pins =
-							<4 31 0x0 0x5>;	/* PE31 GPIO with pull up deglith */
-					};
-
-					pinctrl_usba_vbus: usba_vbus {
-						atmel,pins =
-							<3 29 0x0 0x4>; /* PD29 GPIO with deglitch */
-					};
-				};
-			};
-
-			dbgu: serial@ffffee00 {
-				status = "okay";
-			};
-
-			watchdog@fffffe40 {
 				status = "okay";
 			};
 		};
-
-		usb0: gadget@00500000 {
-			atmel,vbus-gpio = <&pioD 29 0>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usba_vbus>;
-			status = "okay";
-		};
-
-		usb1: ohci@00600000 {
-			status = "okay";
-			num-ports = <3>;
-			atmel,vbus-gpio = <&pioD 25 0
-					   &pioD 26 1
-					   &pioD 27 1
-					  >;
-		};
-
-		usb2: ehci@00700000 {
-			status = "okay";
-		};
-	};
-
-	leds {
-		status = "okay";
-	};
-
-	sound {
-		compatible = "atmel,sama5d3ek-wm8904";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
-
-		atmel,model = "wm8904 @ SAMA5D3EK";
-		atmel,audio-routing =
-			"MICBIAS", "IN1L",
-			"Left Capture Mux", "MICBIAS";
-
-		atmel,ssc-controller = <&ssc0>;
-		atmel,audio-codec = <&wm8904>;
-
-		status = "disabled";
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d3cm.dtsi b/arch/arm/boot/dts/sama5d3cm.dtsi
index 6419a79..4c7ea24 100644
--- a/arch/arm/boot/dts/sama5d3cm.dtsi
+++ b/arch/arm/boot/dts/sama5d3cm.dtsi
@@ -1,11 +1,12 @@
 /*
- * sama5d3cm.dtsi - Device Tree Include file for SAM9A5D3x CPU Module
+ * sama5d3cm.dtsi - Device Tree Include file for SAMA5D3x CPU Module
  *
  *  Copyright (C) 2012 Atmel,
  *                2012 Ludovic Desroches <ludovic.desroches@atmel.com>
  *
  * Licensed under GPLv2 or later.
  */
+/include/ "sama5d3.dtsi"
 
 / {
 	memory {
@@ -24,6 +25,12 @@
 	};
 
 	ahb {
+		apb {
+			macb0: ethernet@f0028000 {
+				phy-mode = "rgmii";
+			};
+		};
+
 		nand0: nand@60000000 {
 			nand-bus-width = <8>;
 			nand-ecc-mode = "hw";
@@ -42,8 +49,7 @@
 
 		d2 {
 			label = "d2";
-			gpios = <&pioE 25 1>;	/* PE25, conflicts with RXD2 */
+			gpios = <&pioE 25 1>;	/* PE25, conflicts with A25, RXD2 */
 		};
-
 	};
 };
diff --git a/arch/arm/boot/dts/sama5d3dm.dtsi b/arch/arm/boot/dts/sama5d3dm.dtsi
new file mode 100644
index 0000000..33e7f3c
--- /dev/null
+++ b/arch/arm/boot/dts/sama5d3dm.dtsi
@@ -0,0 +1,62 @@
+/*
+ * sama5d3dm.dtsi - Device Tree file for SAMA5 display module
+ *
+ *  Copyright (C) 2012 Atmel,
+ *                2012 Ludovic Desroches <ludovic.desroches@atmel.com>
+ *
+ * Licensed under GPLv2 or later.
+ */
+
+/ {
+	ahb {
+		apb {
+			i2c1: i2c@f0018000 {
+				qt1070: keyboard@1b {
+					compatible = "qt1070";
+					reg = <0x1b>;
+					interrupt-parent = <&pioE>;
+					interrupts = <31 0x0>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&pinctrl_qt1070_irq>;
+				};
+			};
+
+			lcd_bus@f0030000 {
+				status = "okay";
+
+				lcd@f0030000 {
+					status = "okay";
+				};
+
+				lcdovl1@f0030140 {
+					status = "okay";
+				};
+
+				lcdovl2@f0030240 {
+					status = "okay";
+				};
+
+				lcdheo1@f0030340 {
+					status = "okay";
+				};
+			};
+
+			adc0: adc@f8018000 {
+				status = "disabled";
+			};
+
+			tsadcc: tsadcc@f8018000 {
+				status = "okay";
+			};
+
+			pinctrl@fffff200 {
+				board {
+					pinctrl_qt1070_irq: qt1070_irq {
+						atmel,pins =
+							<4 31 0x0 0x5>; /* PE31 GPIO with pull up deglith */
+					};
+				};
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/sama5d3mb.dtsi b/arch/arm/boot/dts/sama5d3mb.dtsi
new file mode 100644
index 0000000..35d34f2
--- /dev/null
+++ b/arch/arm/boot/dts/sama5d3mb.dtsi
@@ -0,0 +1,168 @@
+/*
+ * sama5d3mb.dts - Device Tree file for SAMA5D3 mother board
+ *
+ *  Copyright (C) 2012 Atmel,
+ *                2012 Ludovic Desroches <ludovic.desroches@atmel.com>
+ *
+ * Licensed under GPLv2 or later.
+ */
+/include/ "sama5d3cm.dtsi"
+
+/ {
+	model = "Atmel SAMA5D3-EK";
+	compatible = "atmel,sama5ek", "atmel,sama5d3", "atmel,sama5";
+
+	chosen {
+		bootargs = "console=ttyS0,115200 mtdparts=atmel_nand:8M(bootstrap/uboot/kernel)ro,-(rootfs) rw rootfstype=ubifs ubi.mtd=1 root=ubi0:rootfs";
+	};
+
+	ahb {
+		apb {
+			mmc0: mmc@f0000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_mmc0_clk_cmd_dat0 &pinctrl_mmc0_dat1_3 &pinctrl_mmc0_cd>;
+				status = "okay";
+				slot@0 {
+					reg = <0>;
+					bus-width = <4>;
+					cd-gpios = <&pioD 17 0>;
+				};
+			};
+
+			spi0: spi@f0004000 {
+				m25p80@0 {
+					compatible = "atmel,at25df321a";
+					spi-max-frequency = <50000000>;
+					reg = <0>;
+				};
+			};
+
+			/*
+			 * i2c0 conflicts with ISI:
+			 * disable it to allow the use of ISI
+			 * can not enable audio when i2c0 disabled
+			 */
+			i2c0: i2c@f0014000 {
+				wm8904: wm8904@1a {
+					compatible = "wm8904";
+					reg = <0x1a>;
+				};
+			};
+
+			usart1: serial@f0020000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_usart1 &pinctrl_usart1_rts_cts>;
+				status = "okay";
+			};
+
+			isi: isi@f0034000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_isi &pinctrl_isi_pck_as_mck &pinctrl_isi_power &pinctrl_isi_reset>;
+			};
+
+			mmc1: mmc@f8000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_mmc1_clk_cmd_dat0 &pinctrl_mmc1_dat1_3 &pinctrl_mmc1_cd>;
+				status = "okay";
+				slot@0 {
+					reg = <0>;
+					bus-width = <4>;
+					cd-gpios = <&pioD 18 0>;
+				};
+			};
+
+			adc0: adc@f8018000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <
+					&pinctrl_adc0_adtrg
+					&pinctrl_adc0_ad0
+					&pinctrl_adc0_ad1
+					&pinctrl_adc0_ad2
+					&pinctrl_adc0_ad3
+					&pinctrl_adc0_ad4
+					>;
+				status = "okay";
+			};
+
+			macb1: ethernet@f802c000 {
+				phy-mode = "rmii";
+			};
+
+			pinctrl@fffff200 {
+				board {
+					pinctrl_mmc0_cd: mmc0_cd {
+						atmel,pins =
+							<3 17 0x0 0x5>; /* PD17 GPIO with pullup deglitch */
+					};
+
+					pinctrl_mmc1_cd: mmc1_cd {
+						atmel,pins =
+							<3 18 0x0 0x5>; /* PD18 GPIO with pullup deglitch */
+					};
+
+					pinctrl_pck0_as_audio_mck: pck0_as_audio_mck {
+						atmel,pins =
+							<3 30 0x2 0x0>;	/* PD30 periph B */
+					};
+
+					pinctrl_isi_reset: isi_reset-0 {
+						atmel,pins =
+							<4 24 0x0 0x0>;   /* PE24 gpio */
+					};
+
+					pinctrl_isi_power: isi_power-0 {
+						atmel,pins =
+							<4 29 0x0 0x0>; /* PE29 gpio */
+					};
+
+					pinctrl_usba_vbus: usba_vbus {
+						atmel,pins =
+							<3 29 0x0 0x4>; /* PD29 GPIO with deglitch */
+					};
+				};
+			};
+
+			dbgu: serial@ffffee00 {
+				status = "okay";
+			};
+
+			watchdog@fffffe40 {
+				status = "okay";
+			};
+		};
+
+		usb0: gadget@00500000 {
+			atmel,vbus-gpio = <&pioD 29 0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usba_vbus>;
+			status = "okay";
+		};
+
+		usb1: ohci@00600000 {
+			num-ports = <3>;
+			atmel,vbus-gpio = <&pioD 25 0
+					   &pioD 26 1
+					   &pioD 27 1
+					  >;
+			status = "okay";
+		};
+
+		usb2: ehci@00700000 {
+			status = "okay";
+		};
+	};
+
+	sound {
+		compatible = "atmel,sama5d3ek-wm8904";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pck0_as_audio_mck>;
+
+		atmel,model = "wm8904 @ SAMA5D3EK";
+		atmel,audio-routing =
+			"MICBIAS", "IN1L",
+			"Left Capture Mux", "MICBIAS";
+
+		atmel,ssc-controller = <&ssc0>;
+		atmel,audio-codec = <&wm8904>;
+	};
+};
diff --git a/arch/arm/boot/dts/sama5d3pda.dtsi b/arch/arm/boot/dts/sama5d3pda.dtsi
new file mode 100644
index 0000000..5d0dfb1
--- /dev/null
+++ b/arch/arm/boot/dts/sama5d3pda.dtsi
@@ -0,0 +1,76 @@
+/*
+ * sama5d3pda.dtsi - Device Tree file for SAMA5 PDA display module
+ *
+ *  Copyright (C) 2012 Atmel,
+ *                2012 Ludovic Desroches <ludovic.desroches@atmel.com>
+ *
+ * Licensed under GPLv2 or later.
+ */
+
+/ {
+	ahb {
+		apb {
+			i2c1: i2c@f0018000 {
+				qt1070: keyboard@1b {
+					compatible = "qt1070";
+					reg = <0x1b>;
+					interrupt-parent = <&pioE>;
+					interrupts = <30 0x0>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&pinctrl_qt1070_irq>;
+				};
+
+				atmel_mxt_ts@4a {
+					compatible = "atmel,atmel_mxt_ts";
+					reg = <0x4a>;
+					interrupt-parent = <&pioE>;
+					interrupts = <31 0x0>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&pinctrl_mxt_ts>;
+					x_line = <19>;
+					y_line = <11>;
+					x_size = <1024>;
+					y_size = <1024>;
+					blen = <0x0>;
+					threshold = <0x23>;
+					voltage = <3300000>;
+					orient = <0x00>;
+				};
+			};
+
+			lcd_bus@f0030000 {
+				status = "okay";
+
+				lcd@f0030000 {
+					status = "okay";
+				};
+
+				lcdovl1@f0030140 {
+					status = "okay";
+				};
+
+				lcdovl2@f0030240 {
+					status = "okay";
+				};
+
+				lcdheo1@f0030340 {
+					status = "okay";
+				};
+			};
+
+			pinctrl@fffff200 {
+				board {
+					pinctrl_qt1070_irq: qt1070_irq {
+						atmel,pins =
+							<4 30 0x0 0x5>; /* PE31 GPIO with pull up deglith */
+					};
+
+					pinctrl_mxt_ts: mxt_irq {
+						atmel,pins =
+							<4 31 0x0 0x5>; /* PE31 gpio with pullup deglitch */
+					};
+				};
+			};
+		};
+	};
+};
-- 
1.7.10.4

